pkg/service/azure/snapshot.go

```diff
@@ -194,3 +194,61 @@ func (a *azureController) deleteSnapshot(ctx context.Context, req *proto.DeleteS
 	a.logger.Info(ctx, "snapshot deleted", "snapshotid", req.SnapshotId)
 	return &proto.DeleteSnapshotResponse{}, nil
 }
+
+func (a *azureController) copySnapshot(ctx context.Context, req *proto.CopySnapshotRequest) (*proto.CopySnapshotResponse, error) {
+
+	account := req.AccountName
+
+	cred, err := getCredentials(ctx, account)
+	if err != nil {
+		a.logger.Error(ctx, "failed to get the azure credentials", "error", err)
+		return nil, errors.Wrap(err, "copySnapshot")
+	}
+
+	sc, err := getSnapshotClient(cred)
+	if err != nil {
+		a.logger.Error(ctx, "faied to get the snapshot client", "error", err)
+		return nil, errors.Wrap(err, "copySnapshot")
+	}
+	name := fmt.Sprintf("copy-%s", req.SnapshotId)
```

Comment:
Can we pick this from a common place for all providers?

Reply:
there is a follow up todo, we will address it in that PR.